ui/app/src/state/epics/content.ts

```diff
@@ -396,6 +432,81 @@ const content: CrafterCMSEpic[] = [
           })
         );
       })
+    ),
+  // endregion
+  // region Delete Controller
+  (action$, state$) =>
+    action$.pipe(
+      ofType(deleteController.type, deleteTemplate.type),
+      withLatestFrom(state$),
+      switchMap(([{ type, payload }, state]) => {
+        const id = uuid();
+        const { item, onActionSuccess } = payload;
+        const path =
+          type === 'DELETE_CONTROLLER'
+            ? getItemGroovyPath(item, state.contentTypes.byId)
+            : getItemTemplatePath(item, state.contentTypes.byId);
+
+        // path may be empty string if the displayTemplate has not been set for a content type.
+        if (isBlank(path)) {
+          return merge(
+            of(
+              showConfirmDialog({
+                body: getIntl().formatMessage(
+                  itemFailureMessages[type === 'DELETE_CONTROLLER' ? 'controllerNotFound' : 'templateNotFound']
+                )
+              })
+            ),
+            of(popDialog({ id }))
+          );
+        } else {
+          return merge(
+            of(
+              pushDialog({
+                minimized: true,
+                id,
+                status: 'indeterminate',
+                title: getIntl().formatMessage(inProgressMessages.processing),
+                onMaximized: null
+              })
+            ),
+            fetchItemByPath(state.sites.active, path).pipe(
+              map((itemToDelete) =>
+                batchActions([
+                  showDeleteDialog({
+                    items: asArray(itemToDelete),
+                    onSuccess: batchActions([
+                      showDeleteItemSuccessNotification(),
+                      ...(type === 'DELETE_TEMPLATE'
+                        ? [dissociateTemplate({ contentTypeId: item.contentTypeId })]
+                        : []),
+                      closeDeleteDialog(),
+                      ...(onActionSuccess ? [onActionSuccess] : [])
+                    ])
+                  }),
+                  popDialog({ id })
+                ])
+              ),
+              catchAjaxError((error: AjaxError) => {
+                return batchActions([
+                  popDialog({ id }),
+                  ...(error.status === 404
+                    ? [
+                        showConfirmDialog({
+                          body: getIntl().formatMessage(
+                            itemFailureMessages[
+                              type === 'DELETE_CONTROLLER' ? 'controllerNotFound' : 'templateNotFound'
+                            ]
+                          )
+                        })
+                      ]
+                    : [showErrorDialog({ error: error.response ?? error })])
```

Comment:
```suggestion
                  error.status === 404
                    ? 
                        showConfirmDialog({
                          body: getIntl().formatMessage(
                            itemFailureMessages[
                              type === 'DELETE_CONTROLLER' ? 'controllerNotFound' : 'templateNotFound'
                            ]
                          )
                        })
                      
                    : showErrorDialog({ error: error.response ?? error })
```
(format appropriately)